test/e2e/annotations/canary.go

```diff
@@ -1111,8 +1110,9 @@ func TestEvenMainlineCanaryDistribution(f *framework.Framework, host string) {
 
 	assert.Equal(ginkgo.GinkgoT(), 2, len(keys))
 
-	// The implmentation of choice by weight doesn't guarantee exact
-	// number of requests, so verify if request imbalance is within an
-	// acceptable range.
-	assert.LessOrEqual(ginkgo.GinkgoT(), math.Abs(float64(replicaRequestCount[keys[0].String()]-replicaRequestCount[keys[1].String()]))/math.Max(float64(replicaRequestCount[keys[0].String()]), float64(replicaRequestCount[keys[1].String()])), 0.2)
+	// The implementation of choice by weight doesn't guarantee exact
+	// number of requests, so verify if mainline and canary have at
+	// least 40% of requests
```

Comment:
I suggest extracting 200 (line 1092) and 80 (lines 1116 and 1117) as local vars and moving this comment about "40%" (lines 1113-1115) next to those vars, to provide context. Written like this:

* There will be easy connection between 200 and 80; 
* You might also want to actually calculate 80 as 40% instead of mentioning "40%" in the comment; 
* If anyone wants to change either of those numbers, it would be a change in one place instead of 2 or 3.